src/Elastic.Clients.Elasticsearch.Shared/Client/ElasticsearchClient.cs

```diff
@@ -82,17 +92,18 @@ internal ElasticsearchClient(ITransport<IElasticsearchClientSettings> transport)
 	public Serializer SourceSerializer => _transport.Configuration.SourceSerializer;
 	public ITransport<IElasticsearchClientSettings> Transport => _transport;
 
-	private ProductCheckStatus _productCheckStatus;
+	private partial void SetupNamespaces();
+
+	private volatile int _productCheckStatus;
```

Comment:
If all reads and write happen through `Interlocked we should be able to get rid of this volotile keyword.